java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -554,14 +561,19 @@ private void indexLoadedRepo(final RepositoryInfo repo, boolean updateLocal) thr
                     } else {
                         iur.setThreads(1);
                     }
-
-                    NotifyingIndexCreator nic = null;
-                    for (IndexCreator ic : indexingContext.getIndexCreators()) {
-                        if (ic instanceof NotifyingIndexCreator) {
-                            nic = (NotifyingIndexCreator) ic;
-                            break;
-                        }
+                    if (RepositoryPreferences.getIndexDateCutoffFilter() > 0) {
+                        Instant cutoff = ZonedDateTime.now()
+                                .minusYears(RepositoryPreferences.getIndexDateCutoffFilter()).toInstant();
+                        iur.setExtractionFilter(doc -> {
+                            IndexableField date = doc.getField("m"); // usually never null
```

Comment:
Together with an additional import `import static org.apache.maven.index.creator.MinimalArtifactInfoIndexCreator.FLD_LAST_MODIFIED`. This would be more readable:

```suggestion
                            IndexableField date = doc.getField(FLD_LAST_MODIFIED.getKey()); // usually never null
```

Reply:
done